src/gromacs/nblib/tests/testsystems.cpp

```diff
@@ -0,0 +1,289 @@
+ *
+ * GROMACS is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with GROMACS; if not, see
+ * http://www.gnu.org/licenses, or write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA.
+ *
+ * If you want to redistribute modifications to GROMACS, please
+ * consider that scientific software is very special. Version
+ * control is crucial - bugs must be traceable. We will be happy to
+ * consider code for inclusion in the official distribution, but
+ * derived work must not be called official GROMACS. Details are found
+ * in the README & COPYING files - if they are missing, get the
+ * official version at http://www.gromacs.org.
+ *
+ * To help us fund GROMACS development, we humbly ask that you cite
+ * the research papers on the package. Check out http://www.gromacs.org.
+ */
+/*! \internal \file
+ * \brief
+ * This implements nblib test systems
+ *
+ * \author Victor Holanda <victor.holanda@cscs.ch>
+ * \author Joe Jordan <ejjordan@kth.se>
+ * \author Prashanth Kanduri <kanduri@cscs.ch>
+ * \author Sebastian Keller <keller@cscs.ch>
+ */
+#include "gmxpre.h"
+
+#include "testsystems.h"
+
+namespace nblib
+{
+
+struct OwAtom
+{
+    AtomName name = "Ow";
+    Mass     mass = 15.99940;
+    C6       c6   = 0.0026173456;
+    C12      c12  = 2.634129e-06;
+};
+
+struct HAtom
+{
+    AtomName name = "H";
+    Mass     mass = 1.008;
+    C6       c6   = 0;
+    C12      c12  = 0;
+};
+
+struct OMetAtom
+{
+    AtomName name = "OMet";
+    Mass     mass = 15.999;
+    C6       c6   = 0.0022619536;
+    C12      c12  = 1.505529e-06;
+};
+
+struct CMetAtom
+{
+    AtomName name = "CMet";
+    Mass     mass = 15.035; // United atom
+    C6       c6   = 0.0088755241;
+    C12      c12  = 2.0852922e-05;
+};
+
+std::unordered_map<std::string, Charge> Charges{ { "Ow", -0.82 },
+                                                 { "Hw", +0.41 },
+                                                 { "OMet", -0.574 },
+                                                 { "CMet", +0.176 },
+                                                 { "HMet", +0.398 } };
+
+WaterMoleculeBuilder::WaterMoleculeBuilder() : water_("SOL")
+{
+    //! Define Atom Types
+    OwAtom   owAtom;
+    AtomType Ow(owAtom.name, owAtom.mass, owAtom.c6, owAtom.c12);
+    HAtom    hwAtom;
+    AtomType Hw(hwAtom.name, hwAtom.mass, hwAtom.c6, hwAtom.c12);
+
+    //! Add the atoms
+    water_.addAtom(AtomName("Oxygen"), Charges.at("Ow"), Ow);
+    water_.addAtom(AtomName("H1"), Charges.at("Hw"), Hw);
+    water_.addAtom(AtomName("H2"), Charges.at("Hw"), Hw);
+}
+
+Molecule WaterMoleculeBuilder::waterMolecule()
+{
+    addExclusionsFromNames();
```

Comment:
Why isn't this call in the constructor? 

The more important takeaway from this could be: the private variable `water_` is not in a usable state when it is constructed. This multi-phased setup of a molecule will bite us in the posterior at some point.

Encapsulating with another `MoleculeEditor` look like it is adding more bloat at the user layer. We could consider working on this once we have a more stable translation layer and a simulator.

Reply:
This is not in the constructor because in this case I wanted to be able to test different ways of adding exclusions using the same builder.